### PR TITLE
build: Bump minimum supported macOS to 13.0

### PR DESCRIPTION
**Pull request by hebasto**
Running Bitcoin Core on unsupported OSes may expose users to security issues.

macOS Monterey 12 received its final security update ([12.7.6](https://support.apple.com/en-us/100100)) on July 2024. Apple classifies the hardware that can run macOS 12 at most as ["obsolete worldwide"](https://support.apple.com/en-us/102772).

### CODE REVIEW

**Comment by Sjors** [comment]
Concept ACK. See https://github.com/bitcoin/bitcoin/pull/30997#issuecomment-2382352367

**Comment by hebasto** [comment]
> You'll also have to remove `doc/build-osx.md:54:For macOS 11 (Big Sur) and 12 (Monterey) you need to install a more recent version of llvm.`, no?

Thanks! Updated.

**Comment by theuni** [comment]
Concept ACK.

**Comment by Sjors** [comment]
Trying to test if this still works on macOS 13.7 I got stuck. Most likely unrelated to the changes here, but see #31050.

**Comment by hebasto** [comment]
[Qt 6.8 LTS](https://doc.qt.io/qt-6.8/supported-platforms.html#macos) dropped support for macOS 11.

**Review by itornaza — APPROVED** [review]
Concept ACK

**Review by itornaza — APPROVED** [review]
tested ACK 05d25304bc4e0c3058c8ee8a89448ce63ac77304

Run all tests including the functional and extended on macOS 15.0.1 and everything checks out ok. 

Unfortunately, I could not test on macOS previous to 12, as I have converted all my older macs to run Linux.

**Comment by fanquake** [comment]
> Running Bitcoin Core on unsupported OSes may expose users to security issues.
> macOS Big Sur 11 received its last security update ([11.7.10](https://support.apple.com/en-us/106338)) over a year ago.

If this is the reasoning we are going to use, shouldn't this be bumping to `10.13`, given `10.12` is also "unsupported" in terms of security updates?

**Comment by theuni** [comment]
> > Running Bitcoin Core on unsupported OSes may expose users to security issues.
> > macOS Big Sur 11 received its last security update ([11.7.10](https://support.apple.com/en-us/106338)) over a year ago.
> 
> If this is the reasoning we are going to use, shouldn't this be bumping to `10.13`, given `10.12` is also "unsupported" in terms of security updates?

Yes, good point. I didn't realize 12.0 was eol'd. We shouldn't let qt dictate what we support.

Agree that 13 would make more sense, especially considering v29 won't be tagged for quite a while.

**Comment by hebasto** [comment]
> > Running Bitcoin Core on unsupported OSes may expose users to security issues.
> > macOS Big Sur 11 received its last security update ([11.7.10](https://support.apple.com/en-us/106338)) over a year ago.
> 
> If this is the reasoning we are going to use, shouldn't this be bumping to `10.13`, given `10.12` is also "unsupported" in terms of security updates?

You mean "13.0" and "12.0", right?

For the reference, macOS 12 was last updated on 2024-07-29. The three most recent Apple updates did not affect macOS 12.

**Comment by hebasto** [comment]
If we raise the minimum supported macOS version to 13.0, it would mean dropping support for hardware like MacBook Pro (15-inch, 2016) and older, which Apple classifies as ["obsolete worldwide"](https://support.apple.com/en-us/102772).

I'll update this PR shortly.

**Comment by hebasto** [comment]
> > > Running Bitcoin Core on unsupported OSes may expose users to security issues.
> > > macOS Big Sur 11 received its last security update ([11.7.10](https://support.apple.com/en-us/106338)) over a year ago.
> > 
> > 
> > If this is the reasoning we are going to use, shouldn't this be bumping to `10.13`, given `10.12` is also "unsupported" in terms of security updates?
> 
> Yes, good point. I didn't realize 12.0 was eol'd. We shouldn't let qt dictate what we support.
> 
> Agree that 13 would make more sense, especially considering v29 won't be tagged for quite a while.

Done.

The PR description has been updated.

**Comment by m3dwards** [comment]
ACK a0e089a71dc449f4cc177d6eb3050400e63f4b3f

Tested on ARM Mac 14.6

**Comment by itornaza** [comment]
reACK a0e089a71dc449f4cc177d6eb3050400e63f4b3f

**Review by itornaza — APPROVED** [review]
reACK a0e089a71dc449f4cc177d6eb3050400e63f4b3f

**Comment by hebasto** [comment]
My Guix build:
```
aarch64
11dcf8e401c3cf7a911bf08cf801d354b53d8f4266b139625ef12f502050ee4f  guix-build-a0e089a71dc4/output/arm64-apple-darwin/SHA256SUMS.part
d00bfbd05b870fdd4a1b7626bc0e851a88a3b958a1ff78d6822ed85f743c7c8a  guix-build-a0e089a71dc4/output/arm64-apple-darwin/bitcoin-a0e089a71dc4-arm64-apple-darwin-unsigned.tar.gz
fc0dcd590d775f6e565ec6d7a005e525891b90b5afdd1f6d429d9f615792ca2b  guix-build-a0e089a71dc4/output/arm64-apple-darwin/bitcoin-a0e089a71dc4-arm64-apple-darwin-unsigned.zip
a16791436c034c7f84eb9acdc11ddb5a6f95d173af0b73146ac3d9239028be7d  guix-build-a0e089a71dc4/output/arm64-apple-darwin/bitcoin-a0e089a71dc4-arm64-apple-darwin.tar.gz
0d70a5e514d93f44dc02794b604e37ee5a7f30952c2561d79e3141806e2eabff  guix-build-a0e089a71dc4/output/dist-archive/bitcoin-a0e089a71dc4.tar.gz
20b48785ebc782acc4aabd59944e4e923d6c484f000f39d601747f10b3269242  guix-build-a0e089a71dc4/output/x86_64-apple-darwin/SHA256SUMS.part
fd35b25569b68d3df1a50777a60bc370694b61a1073300db113118d583811af8  guix-build-a0e089a71dc4/output/x86_64-apple-darwin/bitcoin-a0e089a71dc4-x86_64-apple-darwin-unsigned.tar.gz
b8e88abb78526371fac9756c5edfd8d0a6ccc44a62a3dd4502337c0390ff6c2c  guix-build-a0e089a71dc4/output/x86_64-apple-darwin/bitcoin-a0e089a71dc4-x86_64-apple-darwin-unsigned.zip
30cb693d61e12dca42c23395b9e29fc31d2002d36d7955e35163bac083b33089  guix-build-a0e089a71dc4/output/x86_64-apple-darwin/bitcoin-a0e089a71dc4-x86_64-apple-darwin.tar.gz
```

**Comment by fanquake** [comment]
Guix Build
```bash
d454a81e02e24f26f4d163f27ee2b16b83c8f1dfcfe0384ab544b704e27fed09  guix-build-a0e089a71dc4/output/aarch64-linux-gnu/SHA256SUMS.part
f5b3771bbb5e2c14e6d91d6a76711df0a9b8ab1b66e118eb6c5aba5686c91902  guix-build-a0e089a71dc4/output/aarch64-linux-gnu/bitcoin-a0e089a71dc4-aarch64-linux-gnu-debug.tar.gz
229f0eef5808a7266ac48b9b02dd84b7a7960353125ed7052510a8014074c0da  guix-build-a0e089a71dc4/output/aarch64-linux-gnu/bitcoin-a0e089a71dc4-aarch64-linux-gnu.tar.gz
0dc7197f0284cf783f7ed5cb4ad15aaceb94588b72d08800cfe35c813b44706d  guix-build-a0e089a71dc4/output/arm-linux-gnueabihf/SHA256SUMS.part
1e7b8584b2b3e741ad51d2c6776ac0a0ab2940aa9ca30f88158fbf911c80dcaa  guix-build-a0e089a71dc4/output/arm-linux-gnueabihf/bitcoin-a0e089a71dc4-arm-linux-gnueabihf-debug.tar.gz
4dc1d6e7113a27842ff032719e3c200e2f49f06de3ba1736980a3c64e5503923  guix-build-a0e089a71dc4/output/arm-linux-gnueabihf/bitcoin-a0e089a71dc4-arm-linux-gnueabihf.tar.gz
11dcf8e401c3cf7a911bf08cf801d354b53d8f4266b139625ef12f502050ee4f  guix-build-a0e089a71dc4/output/arm64-apple-darwin/SHA256SUMS.part
d00bfbd05b870fdd4a1b7626bc0e851a88a3b958a1ff78d6822ed85f743c7c8a  guix-build-a0e089a71dc4/output/arm64-apple-darwin/bitcoin-a0e089a71dc4-arm64-apple-darwin-unsigned.tar.gz
fc0dcd590d775f6e565ec6d7a005e525891b90b5afdd1f6d429d9f615792ca2b  guix-build-a0e089a71dc4/output/arm64-apple-darwin/bitcoin-a0e089a71dc4-arm64-apple-darwin-unsigned.zip
a16791436c034c7f84eb9acdc11ddb5a6f95d173af0b73146ac3d9239028be7d  guix-build-a0e089a71dc4/output/arm64-apple-darwin/bitcoin-a0e089a71dc4-arm64-apple-darwin.tar.gz
0d70a5e514d93f44dc02794b604e37ee5a7f30952c2561d79e3141806e2eabff  guix-build-a0e089a71dc4/output/dist-archive/bitcoin-a0e089a71dc4.tar.gz
41969cf3a3a7b4ed98daf9f4bd008c68dafc9e0da4f5b57819b5468b9a3e3c83  guix-build-a0e089a71dc4/output/powerpc64-linux-gnu/SHA256SUMS.part
3ca12ebb1b4e7137ad310c20e5b65a739eef4d8b08d00d9e781558140b3a80f6  guix-build-a0e089a71dc4/output/powerpc64-linux-gnu/bitcoin-a0e089a71dc4-powerpc64-linux-gnu-debug.tar.gz
d3c1eeef5825646424c981e95a0fdc928503bdd667ae08a116e419184c51fd6d  guix-build-a0e089a71dc4/output/powerpc64-linux-gnu/bitcoin-a0e089a71dc4-powerpc64-linux-gnu.tar.gz
39c9ef33d879ced0256bd02d838204413196d636cef5ba3a608b2c46be0a1a24  guix-build-a0e089a71dc4/output/riscv64-linux-gnu/SHA256SUMS.part
cd416eeece2e8d8630f75866b5509156be378a22874e23a5085d60571a98be46  guix-build-a0e089a71dc4/output/riscv64-linux-gnu/bitcoin-a0e089a71dc4-riscv64-linux-gnu-debug.tar.gz
5099b45320db47873a72d733f4bf85effb2ce20627735458ed8b516aae78615f  guix-build-a0e089a71dc4/output/riscv64-linux-gnu/bitcoin-a0e089a71dc4-riscv64-linux-gnu.tar.gz
20b48785ebc782acc4aabd59944e4e923d6c484f000f39d601747f10b3269242  guix-build-a0e089a71dc4/output/x86_64-apple-darwin/SHA256SUMS.part
fd35b25569b68d3df1a50777a60bc370694b61a1073300db113118d583811af8  guix-build-a0e089a71dc4/output/x86_64-apple-darwin/bitcoin-a0e089a71dc4-x86_64-apple-darwin-unsigned.tar.gz
b8e88abb78526371fac9756c5edfd8d0a6ccc44a62a3dd4502337c0390ff6c2c  guix-build-a0e089a71dc4/output/x86_64-apple-darwin/bitcoin-a0e089a71dc4-x86_64-apple-darwin-unsigned.zip
30cb693d61e12dca42c23395b9e29fc31d2002d36d7955e35163bac083b33089  guix-build-a0e089a71dc4/output/x86_64-apple-darwin/bitcoin-a0e089a71dc4-x86_64-apple-darwin.tar.gz
8c5ab300a7c536e1072f0b202277a7b93d4300a25ab0305e6895db5776ffdecb  guix-build-a0e089a71dc4/output/x86_64-linux-gnu/SHA256SUMS.part
bae293398fe25c846259ee2dbb5853bf6df811b164f7f290e333d8375a8fd48a  guix-build-a0e089a71dc4/output/x86_64-linux-gnu/bitcoin-a0e089a71dc4-x86_64-linux-gnu-debug.tar.gz
87cc2f80cc91e281bcf87e71b4d1d3d804a5e0f12f8db83f0beb4334674f5b39  guix-build-a0e089a71dc4/output/x86_64-linux-gnu/bitcoin-a0e089a71dc4-x86_64-linux-gnu.tar.gz
e1c64c75acea77c614fcdad5e78e2c50d9ff40ee195ea8c6192d61ecd014c277  guix-build-a0e089a71dc4/output/x86_64-w64-mingw32/SHA256SUMS.part
3e7bc074f58551740fafa6a962346b363a051cc90f26a87b0c22654fe5bb9e59  guix-build-a0e089a71dc4/output/x86_64-w64-mingw32/bitcoin-a0e089a71dc4-win64-debug.zip
5eced49cd17792112a1ad580b7d7d861808549c0242773c20412f3878211caea  guix-build-a0e089a71dc4/output/x86_64-w64-mingw32/bitcoin-a0e089a71dc4-win64-setup-unsigned.exe
92273a7aa4a99a5cbfd64284c4d6739d68646f8c35d3b8747b4d132b3118bb4e  guix-build-a0e089a71dc4/output/x86_64-w64-mingw32/bitcoin-a0e089a71dc4-win64-unsigned.tar.gz
d3d26599cdf3497862e5c218061d81d50f4bfa908826a0f6b5ced1a1dca4946e  guix-build-a0e089a71dc4/output/x86_64-w64-mingw32/bitcoin-a0e089a71dc4-win64.zip
```

File: doc/build-osx.md [hunk 1]
```diff
@@ -51,8 +51,6 @@ To install, run the following from your terminal:
 brew install cmake boost pkg-config libevent
 ```
 
-For macOS 11 (Big Sur) and 12 (Monterey) you need to install a more recent version of llvm.
-
 ``` bash
 brew install llvm
 ```
```
[the remark this answers is not in the document]
Comment (Sjors) [reply doc/build-osx.md:56]: Yes that should be removed, along with the rest of the paragraph below.

[the remark this answers is not in the document]
Comment (fanquake) [reply doc/build-osx.md:56]: Removed in #31099.

File: doc/release-notes-empty-template.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ Compatibility
 ==============
 
 Bitcoin Core is supported and extensively tested on operating systems
-using the Linux Kernel 3.17+, macOS 11.0+, and Windows 7 and newer. Bitcoin
+using the Linux Kernel 3.17+, macOS 13.0+, and Windows 7 and newer. Bitcoin
```
[the remark this answers is not in the document]
Comment (hebasto) [reply doc/release-notes-empty-template.md:46]: > Do you plan to open a follow-up to apply the similar change to Windows?

Yes. When I find a proper way to specify the minimum required OS version. The currently used `--major-subsystem-version` and `--minor-subsystem-version` flags do not work for me. And Microsoft docs are not very helpful. So any hints are much appreciated.

[the remark this answers is not in the document]
Comment (fanquake) [reply doc/release-notes-empty-template.md:46]: Yea, and this is already the case with the current binary. See #31172.

**Pull request merged**